### ISSUE TITLE
High-level forward thinking of what we want the Mail app to be

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/mail#65]
Right now the Mail app is an ok IMAP client. It has some nice integration with other Nextcloud apps, pretty standard stuff.

Now how can we make that even better and even simpler? Just some brainstorming, also partially things which have been mentioned in other issues, but here as a more coherent combination. Please don’t take this as a bashing – I’m a contributor and this is just me trying to take a step back and check what a person would expect. :)

- :star: **Favorites** are a separate folder right now. It seems like an afterthought. Sure, other mail apps do the same, but shouldn’t we try to innovate here? Sorting the favorites of a specific view (say, the inbox) up top of that view would be nice for example so you are reminded to deal with them. It’s a way of »pinning« conversations.
- :airplane: **Sent** emails are a separate folder too. Of course this is because every other email app does it like this … but almost no other _communication_ system like SMS, Facebook, Whatsapp, Slack, IRC, etc does it like this. It’s all one stream. Here #17 (conversations) comes in of course.
- :page_facing_up: **Reading** is a bit bland and also overwhelming. Title at the top, but names without avatars (we definitely need to add those). But then there’s big useless parts like the long previous replies, signatures and mailing list footers which are really not something I want to ever see.
- :pencil: **Writing** an email currently feels a bit basic. There’s the whole stuff from the previous mail which is just dumped there, the typeface is small, there are many boxes, there are no faces of people. We need to enhance this a bit.
- :o: **Focus** can be increased by doing something we have thought about for other apps (like Calendar and Maps) for some time now: Hiding the left navigation bar by default even in desktop view, making it expandable via the menu icon. I at least spend most of my time in the unified inbox and mostly go to the other folders to delete things or check on sent mails (see issue above).
- :boom: **Stability** sometimes varies. Right now and the last few weeks I had many small issues like mails not loading at all, send button not working, the reply to/cc being filled out wrongly and especially server errors in the unified inbox. It’s all things which we absolutely need to get right and which always have priority over new features.
- :question: Probably more … ?

@nextcloud/mail what do you think? The conference would be a good place to talk about how we can focus on the essentials and make sure that’s the absolute center of the app. :) And then polish it reeeally good.

**Comment by ChristophWurst** [comment]
:+1: sounds very good

**Comment by ChristophWurst** [comment]
What we should not do again is pull in big features that we cannot handle as they do not work as nicely as they should. We're still facing **a lot** of problems with the unified inbox hackery, and that feature was merged ages ago.

But I agree, on a higher level we should try to create something useful for the users, not just re-invent the wheel and create another crappy web-based IMAP client…

> :boom: Stability sometimes varies. Right now and the last few weeks I had many small issues like mails not loading at all, send button not working, the reply to/cc being filled out wrongly and especially server errors in the unified inbox. It’s all things which we absolutely need to get right and which always have priority over new features.

To master this one we could try to automate testing as far as possible, most importantly with js unit tests.

**Comment by pierreozoux** [comment]
- :abcd: keyboard shortcut
 - Possibility to move emails into different folders.

I think first priority should be to make it stable and nice to use (I still can't use it on everyday basis, where I can with thunderbird, or I could with gmail which I don't use any more).
And then, yes, all your idea are really nice :) Looking forward 1.0 milestone!

**Comment by ChristophWurst** [comment]
Let's try GitHub's new feature and create a project for this :-) https://github.com/nextcloud/mail/projects

**Comment by ChristophWurst** [comment]
done: https://github.com/nextcloud/mail/projects/2

**Comment by plastikschnitzer** [comment]
I am also on nextcloud now and my preferred projects for the mail app are stability, signature and folder management. I can help you with beta-testing in nextcloud and owncloud. Have fun at the NC conference!

**Comment by brjhaverkamp** [comment]
my 2 cts: Move mail into other folders (drag and drop preffered) and folder management.

**Comment by jancborchardt** [comment]
@brjhaverkamp there’s a pull request open for moving mails: https://github.com/nextcloud/mail/pull/90 any review and contribution is much appreciated. :)

**Comment by Bugsbane** [comment]
Something I find really useful in Facebook / Twitter / Signal is how easy it is to basically just click on a contact and see all of my conversation with that person in one single, scrollable stream, where in most email clients I find myself having to search for specific emails a lot.

**Comment by jancborchardt** [comment]
Yep. That’s one idea we could have in the sidebar. Folder-like filters in the sidebar which are simply saved searches for the most contacted people. :) The icon of the folder would be the avatar of that person.

**Comment by Bugsbane** [comment]
Ok, and presumably if you clicked on that person, you'd see a single stream with the contents of all of your conversations? Because that was the most important part of what I was saying. Not so much that it's easy to find contacts, but that the things you've said to each other are easily viewable as a record of your conversation, rather than being split up into a million fragmented emails.

**Comment by jancborchardt** [comment]
@Bugsbane yes, exactly what you said. Show all messages with them in chronological order.

Of course the main problem here is the same as with #17 (Grouping of message threads / conversations), in that we need to query several folders via IMAP for this. At least Inbox and Sent anyway, to get both the messages from you and the other person.

**Comment by pieter-groeneweg** [comment]
I stumbled upon this through filter "sort order". As that would be nice to have, sort by sender/receiver, date, size, ....

**Comment by Friptick** [comment]
> I stumbled upon this through filter "sort order". As that would be nice to have, sort by sender/receiver, date, size, ....

Unfortunately my feature request (#679) for this has just been closed, partly for "lack of interest" (fair enough).

A mail list-view is just a table. Basic table UX means sortability. Surely before getting into fancy features like the ones suggested in this thread, it would not be too difficult to add this basic one?

**Comment by fvzwieten** [comment]
Filters (rules) and folders like with gmail would be awesome.

**Comment by fvzwieten** [comment]
A better UX for selecting mail addresses in the To and CC fields. The current one is highly confusing and thus bad UX.

**Comment by dodedodo** [comment]
I think the UI / Theming needs more separation or shading. Right now, mail is just a white page with an overload of text, and no easy way to focus on the stuff you want to know. Call it visual grepping if you want.

I can't really explain it that well, so I fiddled around some with the devtools and made some examples. These are by no means design proposals! 

---
<summary> 
In the left example it's a lot easier to find the 'drafts for mail account X' for example. 
<details>
Don't mind the ugly colorscheme :P.
<img src=https://user-images.githubusercontent.com/3252036/124490526-7dd59a80-ddb2-11eb-8aa8-a1f3c034d9c2.png />
</details>
</summary>

<summary> 
In the bottom example it's clear which header corresponds to what. There's a tread, and 2 messages. Instead of a header and then another header saying essentially the same thing. 
It also visually separates the messages and email pane. 
<details>
Don't mind the ugly colorscheme :P.
<img src=https://user-images.githubusercontent.com/3252036/124496564-de1c0a80-ddb9-11eb-8762-ed0f0c040317.png />
</details>
</summary>

**Comment by ChristophWurst** [comment]
@dodedodo good timing. We plan to have a closer look on the app's design soon to improve those things. Ref https://github.com/orgs/nextcloud/projects/45#card-63246375. You are very welcome to join our call(s) where we'll discuss those topics but also in github tickets if you prefer async communication :)
